### PR TITLE
chore(2916): refactor stream handler for phishing warning and provider stream

### PR DESCRIPTION
**Pull request by DDDDDanica**
## **Description**
As part of https://github.com/MetaMask/MetaMask-planning/issues/2916, it is cleaner to extract the stream out of contentscript file and re-use the logic of phishing stream before implement a new stream.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26441?quickstart=1)

## **Related issues**

Part of https://github.com/MetaMask/MetaMask-planning/issues/2916
## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/constants/stream.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export const METAMASK_INPAGE = 'metamask-inpage';
+export const PHISHING_WARNING_PAGE = 'metamask-phishing-warning-page';
+
+// stream channels
+export const METAMASK_PROVIDER = 'metamask-provider';
+export const METAMASK_COOKIE_HANDLER = 'metamask-cookie-handler';
+export const PHISHING_SAFELIST = 'metamask-phishing-safelist';
+export const PHISHING_STREAM = 'phishing';
+
+// TODO:LegacyProvider: Delete
```
Comment (seaona) [inline app/scripts/constants/stream.ts:12]:
```suggestion
// For more information about these legacy streams, see here:
// https://github.com/MetaMask/metamask-extension/issues/15491
// TODO:LegacyProvider: Delete
```

maybe we keep the above comment, for more context?

File: app/scripts/constants/stream.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// contexts
+export const CONTENT_SCRIPT = 'metamask-contentscript';
+export const METAMASK_INPAGE = 'metamask-inpage';
+export const PHISHING_WARNING_PAGE = 'metamask-phishing-warning-page';
+
+// stream channels
+export const METAMASK_PROVIDER = 'metamask-provider';
+export const METAMASK_COOKIE_HANDLER = 'metamask-cookie-handler';
```
Comment (seaona) [inline app/scripts/constants/stream.ts:8]:
```suggestion
export const METAMASK_COOKIE_HANDLER = 'metamask-cookie-handler';
export const METAMASK_PROVIDER = 'metamask-provider';
```
just a nit: swap the 2 so it's ordered alphabetically

File: app/scripts/streams/phishing-stream.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
```
Comment (seaona) [inline app/scripts/streams/phishing-stream.ts:1]: typescript, nice!! :heart_eyes:

**Review by seaona — DISMISSED** [review]
Despite not being super familiar with this part of the code, the refactor looks good on my side: I've checked that the new code is consistent with the old code, which from the description, should be expected. Just added a couple of small non-blocking nits

**Review by seaona — COMMENTED** [review]
hi there :wave: as per the refactor (moving logic in other files) looks good, but about the new changes introduced recently (ie `phishingPageMux.ignoreStream`) I'm not sure, as I'm not familiar with that code, so possibly someone with experience with that part is a better suit for reviewing this part

**Review by NiranjanaBinoy — DISMISSED** [review]

**Review by NiranjanaBinoy — DISMISSED** [review]

**Review by vthomas13 — APPROVED** [review]

**Review by itsyoboieltr — APPROVED** [review]

**Pull request merged**